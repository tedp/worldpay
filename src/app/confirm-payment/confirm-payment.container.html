<mat-card>
  <mat-card-header>
    <mat-card-title>Payment confirmation</mat-card-title>
  </mat-card-header>
  <mat-card-content  *ngIf="paymentRquest$ | async as paymentRquest">
      <p class="mat-body-2">Narrative: {{paymentRquest.instruction.narrative}}</p>
      <p class="mat-body-2">currency: {{paymentRquest.instruction.value.currency}}</p>
      <p class="mat-body-2">Amount: {{paymentRquest.instruction.value.amount}}</p>
      <p class="mat-body-2">Card type: {{paymentRquest.instruction.paymentInstrument.type}}</p>
      <p class="mat-body-2">Card number: {{paymentRquest.instruction.paymentInstrument.cardNumber}}</p>
      <p class="mat-body-2">Card holder name: {{paymentRquest.instruction.paymentInstrument.cardHolderName}}</p>
  </mat-card-content>
  <mat-card-actions *ngIf="authorizationResponse$ | async as authorizationResponse">
    <button mat-raised-button type="button" color="primary" (click)="confirmPayment()">Confirm payment</button>
    <button mat-raised-button type="button" color="secondary" (click)="cancelPayment(authorizationResponse)">Cancel payment</button>
  </mat-card-actions>
</mat-card>
