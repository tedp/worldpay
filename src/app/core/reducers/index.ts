import { ActionReducerMap, MetaReducer } from '@ngrx/store';
import { environment } from '../../../environments/environment';
import { PaymentsState, paymentReducer } from './payments.reducer';

export interface AppState {
  readonly payments: PaymentsState;
}

export const reducers: ActionReducerMap<AppState> = {
  payments: paymentReducer
};

export const metaReducers: MetaReducer<AppState>[] = !environment.production ? [] : [];

export * from './payments.reducer';

