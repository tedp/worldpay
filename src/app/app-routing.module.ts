import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import { TakePaymentContainer } from './take-payment/take-payment.container';
import { ConfirmPaymentContainer } from './confirm-payment/confirm-payment.container';
import { PaymentCancelledContainer } from './payment-cancelled/payment-cancelled.container';
import { PaymentSuccessContainer } from './payment-success/payment-success.container';

const routes: Routes = [
  { path: 'take-payment', component: TakePaymentContainer},
  { path: 'confirm-payment', component: ConfirmPaymentContainer},
  { path: 'payment-success', component: PaymentSuccessContainer},
  { path: 'payment-cancelled', component: PaymentCancelledContainer},
  { path: '',
    redirectTo: '/take-payment',
    pathMatch: 'full'
  },
 ];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
