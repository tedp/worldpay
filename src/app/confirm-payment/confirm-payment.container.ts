import { Component, OnInit } from '@angular/core';
import { Store } from '@ngrx/store';
import { AppState } from '../core/reducers';
import { Router } from '@angular/router';
import { Observable } from 'rxjs';
import { getAuthorizationResponse, getPaymentRequest } from '../core/selectors';
import { HypermediaResult, PaymentRequest } from '../core/model/model';
import { CancelAction } from '../core/actions/payments.actions';

@Component({
  selector: 'app-confirm-payment',
  templateUrl: './confirm-payment.container.html',
  styleUrls: ['./confirm-payment.container.scss']
})
export class ConfirmPaymentContainer implements OnInit {

  paymentRquest$: Observable<PaymentRequest>;
  authorizationResponse$: Observable<HypermediaResult>;

  constructor(
    private store: Store<AppState>,
    private router: Router) {
      this.paymentRquest$ = this.store.select(getPaymentRequest);
      this.authorizationResponse$ = this.store.select(getAuthorizationResponse);
  }

  ngOnInit() {
  }

  confirmPayment() {
    // Dispatch action to settle payment
    this.router.navigate(['/payment-success']);
  }

  cancelPayment(authorizationResponse: HypermediaResult) {
    this.store.dispatch(new CancelAction({ authorizationResponse }));
    this.router.navigate(['/payment-cancelled']);
  }

}
