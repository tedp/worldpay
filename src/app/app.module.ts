import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';

import { AppRoutingModule } from './app-routing.module';
import { AppComponent } from './app.component';
import { EffectsModule } from '@ngrx/effects';
import { PaymentsEffects } from './core/effects/payments.effects';
import { StoreModule } from '@ngrx/store';
import { reducers, metaReducers } from './core/reducers';
import { TakePaymentContainer } from './take-payment/take-payment.container';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatInputModule, MatButtonModule, MatSelectModule, MatRadioModule, MatCardModule } from '@angular/material';
import { StoreDevtoolsModule } from '@ngrx/store-devtools';
import { environment } from '../environments/environment';
import { ConfirmPaymentContainer } from './confirm-payment/confirm-payment.container';
import { PaymentSuccessContainer } from './payment-success/payment-success.container';
import { PaymentCancelledContainer } from './payment-cancelled/payment-cancelled.container';


@NgModule({
  declarations: [
    AppComponent,
    TakePaymentContainer,
    ConfirmPaymentContainer,
    PaymentSuccessContainer,
    PaymentCancelledContainer
  ],
  imports: [
    BrowserModule,
    HttpClientModule,
    AppRoutingModule,
    ReactiveFormsModule,
    EffectsModule.forRoot([PaymentsEffects]),
    StoreModule.forRoot(reducers, { metaReducers }),
    NoopAnimationsModule,
    MatInputModule,
    MatButtonModule,
    MatSelectModule,
    MatRadioModule,
    MatCardModule,
    StoreDevtoolsModule.instrument({ maxAge: 25, logOnly: environment.production })
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
