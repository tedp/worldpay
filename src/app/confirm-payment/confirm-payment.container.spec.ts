import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ConfirmPaymentContainer } from './confirm-payment.container';
import { AppModule } from '../app.module';
import { APP_BASE_HREF } from '@angular/common';

describe('ConfirmPaymentComponent', () => {
  let component: ConfirmPaymentContainer;
  let fixture: ComponentFixture<ConfirmPaymentContainer>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [ AppModule ],
      providers:
      [
        {provide: APP_BASE_HREF, useValue : 'confirm-payment' }
      ]

    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ConfirmPaymentContainer);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  // More test should be added here along the lines of take-payment-container.spec.ts
});
