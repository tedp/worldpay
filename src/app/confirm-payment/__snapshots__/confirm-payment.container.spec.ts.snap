// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPaymentComponent should match snapshot 1`] = `
<app-confirm-payment
  authorizationResponse$={[Function Store]}
  paymentRquest$={[Function Store]}
  router={[Function Router]}
  store={[Function Store]}
>
  <mat-card
    class="mat-card"
  >
    <mat-card-header
      class="mat-card-header"
    >
      <div
        class="mat-card-header-text"
      >
        <mat-card-title
          class="mat-card-title"
        >
          Payment confirmation
        </mat-card-title>
      </div>
    </mat-card-header>
    
    
  </mat-card>
</app-confirm-payment>
`;
